packages/headless/src/controllers/insight/search-status/headless-insight-status.ts

```diff
@@ -0,0 +1,28 @@
+import {InsightEngine} from '../../../app/insight-engine/insight-engine';
+import {Controller} from '../../controller/headless-controller';
+import {
+  CoreStatusState,
+  buildCoreStatus,
+} from '../../core/status/headless-core-status';
+
+/**
+ * The `InsightStatus` controller provides information on the status of the search.
+ */
+export interface InsightStatus extends Controller {
+  state: InsightStatusState;
+}
+
+/**
+ * A scoped and simplified part of the headless state that is relevant to the `InsightStatus` controller.
+ */
+export interface InsightStatusState extends CoreStatusState {}
+
+/**
+ * Creates a `InsightStatus` controller instance.
+ *
+ * @param engine - The headless engine.
+ * @returns A `InsightStatus` controller instance.
+ */
+export function buildInsightStatus(engine: InsightEngine): InsightStatus {
```

Comment:
Not sure about why we need  the `buildInsightStatus` function that simply  calls the `buildCoreStatus` without any additions? will this controller contain additional logic specific to the insight panel in the future?

Reply:
I'd say the main reason is to enforce the engine type.

Suppose we expose the `buildCoreStatus(engine: CoreEngine)` function directly in the Insight headless bundle. Then, adding any insight-specific logic in the future would cause a breaking change because we'd need to change the type of the `engine` parameter to `InsightEngine` which is more specific than `CoreEngine`. Having the `buildInsightStatus` function give us more freedom to implement insight-specific functionalities in the future without breaking the API.